gflownet/envs/amp.py

```diff
@@ -202,34 +211,38 @@ def true_density(self, max_states=1e6):
     def state2oracle(self, state: List = None):
         return "".join(self.state2readable(state))
 
-    def statebatch2oracle(self, state_batch: List[List]):
-        """
-        Prepares a sequence in "GFlowNet format" for the oracles.
-
-        Args
-        ----
-        state_list : list of lists
-            List of sequences.
-        """
-        state_oracle = [self.state2oracle(state) for state in state_batch]
-        return state_oracle
-
     def get_max_traj_len(
         self,
     ):
         return self.max_seq_length / self.min_word_len + 1
 
+    def statebatch2oracle(
+        self, states: List[TensorType["max_seq_length"]], bos_idx=None
+    ) -> List[str]:
+        state_oracle = []
+        for state in states:
+            if state[-1] == self.padding_idx:
+                state = state[: torch.where(state == self.padding_idx)[0][0]]
+            if bos_idx is not None and state[0] == bos_idx:
+                state = state[1:-1]
+            state_numpy = state.detach().cpu().numpy()
+            state_oracle.append(self.state2oracle(state_numpy))
+        return state_oracle
+    
     def statetorch2oracle(
-        self, states: TensorType["batch", "state_dim"]
-    ) -> TensorType["batch", "state_proxy_dim"]:
+        self, states: TensorType["batch_dim", "max_seq_length"], bos_idx=None
+    ) -> List[str]:
         state_oracle = []
         for state in states:
-            if state[-1] == self.invalid_state_element:
-                state = state[: torch.where(state == self.invalid_state_element)[0][0]]
+            if state[-1] == self.padding_idx:
+                state = state[: torch.where(state == self.padding_idx)[0][0]]
+            if bos_idx is not None and state[0] == bos_idx:
+                state = state[1:-1]
             state_numpy = state.detach().cpu().numpy()
             state_oracle.append(self.state2oracle(state_numpy))
         return state_oracle
```

Comment:
This is nearly identical to `statebatch2oracle` so I would make one use the other.